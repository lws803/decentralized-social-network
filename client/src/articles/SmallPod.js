import React from "react";

import styled from "styled-components";

export default function SmallPod(props) {
  return (
    <PodContainer
      onClick={props.onClick}
      style={{
        width: `${props.size.width}px`,
      }}
    >
      {props.coverPhoto && <CoverPhoto src={props.coverPhoto} />}
      <Title>{props.title}</Title>
    </PodContainer>
  );
}

const Title = styled.div`
  font-weight: heavy;
  font-size: 20px;
  margin-top: 10px;
  font-family: Georgia;
  word-wrap: break-word;
  line-height: 1.35em;
  padding: 5px;
`;

const PodContainer = styled.div`
  border: solid;
  height: auto;
  clear: both;
  display: inline-block;
  cursor: pointer;
`;

const CoverPhoto = styled.img`
  object-fit: cover;
  width: 100%;
`;
