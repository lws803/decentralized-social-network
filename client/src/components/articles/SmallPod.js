/** @jsx jsx */

import styled from "styled-components";
import { css, jsx } from "@emotion/core";
import PropTypes from "prop-types";

import LazyImage from "../common/LazyImage";

const SmallPod = props => {
  return (
    <PodContainer
      onClick={props.onClick}
      css={css`
        width: ${props.size.width}px;
      `}
      isLoaded
    >
      {props.coverPhoto && (
        <LazyImage
          src={props.coverPhoto}
          css={css`
            object-fit: cover;
            width: 100%;
          `}
        />
      )}
      <Title>{props.title}</Title>
    </PodContainer>
  );
};

SmallPod.propTypes = {
  onClick: PropTypes.func,
  size: PropTypes.object,
  coverPhoto: PropTypes.string,
};

const Title = styled.div`
  font-weight: heavy;
  font-size: 20px;
  margin-top: 10px;
  font-family: Georgia;
  word-wrap: break-word;
  line-height: 1.35em;
  padding: 5px;
`;

const PodContainer = styled.div`
  @keyframes slowlyAppear {
    0% {
      opacity: 0;
    }
    100% {
      opacity: 1;
    }
  }
  margin: 10px;
  border: solid;
  height: auto;
  display: inline-block;
  cursor: pointer;
  animation: 0.25s ease-in-out 0s 1 slowlyAppear;
`;

export default SmallPod;
