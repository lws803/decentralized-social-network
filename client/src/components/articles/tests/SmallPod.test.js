import React from "react";
import renderer from "react-test-renderer";

import SmallPod from "../SmallPod";

it("renders correctly", () => {
  const smallPodContainer = renderer
    .create(<SmallPod size={{ width: 200 }} title="Some random title" />)
    .toJSON();
  expect(smallPodContainer).toMatchSnapshot();
});
