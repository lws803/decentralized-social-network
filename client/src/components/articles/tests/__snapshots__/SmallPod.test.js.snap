// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-AxirZ btopKa css-qix29n-SmallPod"
>
  <div
    className="sc-AxjAm htHRIb"
  >
    Some random title
  </div>
</div>
`;
